packages/openid4vc/src/shared/utils.ts

```diff
@@ -0,0 +1,103 @@
+import type { OpenId4VcJwtIssuer } from './models'
+import type { AgentContext, JwaSignatureAlgorithm, Key } from '@aries-framework/core'
+import type { DIDDocument, SigningAlgo, SuppliedSignature } from '@sphereon/did-auth-siop'
+
+import {
+  AriesFrameworkError,
+  DidsApi,
+  TypedArrayEncoder,
+  getKeyFromVerificationMethod,
+  getJwkClassFromKeyType,
+  SignatureSuiteRegistry,
+} from '@aries-framework/core'
+
+/**
+ * Returns the JWA Signature Algorithms that are supported by the wallet.
+ *
+ * This is an approximation based on the supported key types of the wallet.
+ * This is not 100% correct as a supporting a key type does not mean you support
+ * all the algorithms for that key type. However, this needs refactoring of the wallet
+ * that is planned for the 0.5.0 release.
+ */
+export function getSupportedJwaSignatureAlgorithms(agentContext: AgentContext): JwaSignatureAlgorithm[] {
+  const supportedKeyTypes = agentContext.wallet.supportedKeyTypes
+
+  // Extract the supported JWS algs based on the key types the wallet support.
+  const supportedJwaSignatureAlgorithms = supportedKeyTypes
+    // Map the supported key types to the supported JWK class
+    .map(getJwkClassFromKeyType)
+    // Filter out the undefined values
+    .filter((jwkClass): jwkClass is Exclude<typeof jwkClass, undefined> => jwkClass !== undefined)
+    // Extract the supported JWA signature algorithms from the JWK class
+    .flatMap((jwkClass) => jwkClass.supportedSignatureAlgorithms)
+
+  return supportedJwaSignatureAlgorithms
+}
+
+export async function getSphereonSuppliedSignatureFromJwtIssuer(
+  agentContext: AgentContext,
+  jwtIssuer: OpenId4VcJwtIssuer
+): Promise<SuppliedSignature> {
+  let key: Key
+  let alg: string
+  let kid: string | undefined
+  let did: string | undefined
+
+  if (jwtIssuer.method === 'did') {
+    const didsApi = agentContext.dependencyManager.resolve(DidsApi)
+    const didDocument = await didsApi.resolveDidDocument(jwtIssuer.didUrl)
+    const verificationMethod = didDocument.dereferenceKey(jwtIssuer.didUrl, ['authentication'])
+
+    // get the key from the verification method and use the first supported signature algorithm
+    key = getKeyFromVerificationMethod(verificationMethod)
+    const _alg = getJwkClassFromKeyType(key.keyType)?.supportedSignatureAlgorithms[0]
+    if (!_alg) throw new AriesFrameworkError(`No supported signature algorithms for key type: ${key.keyType}`)
+
+    alg = _alg
+    kid = verificationMethod.id
+    did = verificationMethod.controller
+  } else {
+    throw new AriesFrameworkError("Unsupported jwt issuer method. Only 'did' is supported.")
```

Comment:
```suggestion
    throw new AriesFrameworkError(`Unsupported jwt issuer method ('${jwtIssuer.method}'). Only 'did' is supported.`)
```

Reply:
The PAIN of typescript making it 'never' and thus throwing an error here :(

Basically with TS this code can never be reached, but in JS land it can ...